mipengine/controller/algorithm_executor_nodes.py

```diff
@@ -0,0 +1,346 @@
+        return self._node_tasks_handler.get_table_data(table_name.full_table_name)
+
+    def create_table(self, command_id: str, schema: TableSchema) -> NodeTable:
+        return NodeTable(
+            self._node_tasks_handler.create_table(
+                context_id=self.context_id,
+                command_id=command_id,
+                schema=schema,
+            )
+        )
+
+    # VIEWS functionality
+    def get_views(self) -> List[NodeTable]:
+        result = self._node_tasks_handler.get_views(context_id=self.context_id)
+        return [NodeTable(table_name) for table_name in result]
+
+    # TODO: this is very specific to mip, very inconsistent with the rest, has to
+    # be abstracted somehow
+    def create_pathology_view(
+        self,
+        command_id: str,
+        pathology: str,
+        columns: List[str],
+        filters: List[str],
+    ) -> NodeTable:
+
+        result = self._node_tasks_handler.create_pathology_view(
+            context_id=self.context_id,
+            command_id=command_id,
+            pathology=pathology,
+            columns=columns,
+            filters=filters,
+        )
+        return NodeTable(result)
+
+    # MERGE TABLES functionality
+
+    def get_merge_tables(self) -> List[NodeTable]:
+        result = self._node_tasks_handler.get_merge_tables(context_id=self.context_id)
+        return [NodeTable(table_name) for table_name in result]
+
+    def create_merge_table(self, command_id: str, table_names: List[str]) -> NodeTable:
+        result = self._node_tasks_handler.create_merge_table(
+            context_id=self.context_id,
+            command_id=command_id,
+            table_names=table_names,
+        )
+        return NodeTable(result)
+
+    # REMOTE TABLES functionality
+
+    def get_remote_tables(self) -> List[str]:
+        return self._node_tasks_handler.get_remote_tables(context_id=self.context_id)
+
+    def create_remote_table(
+        self, table_name: str, table_schema: TableSchema, native_node: "_Node"
+    ):
+
+        monetdb_socket_addr = native_node.node_address
+        self._node_tasks_handler.create_remote_table(
+            table_name=table_name,
+            table_schema=table_schema,
+            original_db_url=monetdb_socket_addr,
+        )
+
+    # UDFs functionality
+    def queue_run_udf(
+        self,
+        command_id: str,
+        func_name: str,
+        positional_args: UDFPosArguments,
+        keyword_args: UDFKeyArguments,
+        use_smpc: bool = False,
+    ) -> IQueuedUDFAsyncResult:
+        return self._node_tasks_handler.queue_run_udf(
+            context_id=self.context_id,
+            command_id=command_id,
+            func_name=func_name,
+            positional_args=positional_args,
+            keyword_args=keyword_args,
+            use_smpc=use_smpc,
+        )
+
+    # TODO Controller integration with SMPC
+    # use_smpc should be coming from the request
+
+    @abstractmethod
+    def get_queued_udf_result(
```

Comment:
get_queued_udf_result is alreday defined as an abstract method in the parent interface _INode

Reply:
I want this method to throw a `NotImplementedError` not just `pass`.

Do you want me to add `NotImplementedError` on the interface or leave it like it is now?

Reply:
I am just saying that you are overriding an abstract method of the parent class( actually an interface) with an abstract method.

Reply:
You are right, an exception is raised anyway when initializing a class that doesn't implement the abstract methods.

Removed it.